app/src/main/java/com/babylon/wallet/android/presentation/transaction/analysis/processor/TransactionTypeExtensions.kt

```diff
@@ -690,40 +668,88 @@ fun List<Transferable>.toAccountWithTransferableResources(
     }
 }
 
+suspend fun ExecutionSummary.allOwnedAccounts(getProfileUseCase: GetProfileUseCase): List<Network.Account> {
+    val involvedAccountAddresses = accountWithdraws.keys + accountDeposits.keys
+    return getProfileUseCase.accountsOnCurrentNetwork().filter {
+        involvedAccountAddresses.contains(it.address)
+    }
+}
+
 fun ExecutionSummary.toWithdrawingAccountsWithTransferableAssets(
     involvedAssets: List<Asset>,
-    allOwnedAccounts: List<Network.Account>
-) = accountWithdraws.map { withdrawEntry ->
-    withdrawEntry.value.map { resource ->
-        if (resource.isNewlyCreated(summary = this)) {
-            resource.toNewlyCreatedTransferableAsset(resource.newlyCreatedMetadata(summary = this))
+    allOwnedAccounts: List<Network.Account>,
+    aggregateByResourceAddress: Boolean = false // for now I turned aggregation off to not introduce any bugs
+): List<AccountWithTransferableResources> {
+    return accountWithdraws.map { withdrawEntry ->
+        if (aggregateByResourceAddress) {
+            withdrawEntry.value.groupBy { it.resourceAddress }.map { indicatorsPerResource ->
+                val resources = indicatorsPerResource.value
+                when (val first = resources.first()) {
+                    is ResourceIndicator.Fungible -> {
+                        val aggregateAmount = resources.map { it.amount }.sumOf { it }
+                        if (first.isNewlyCreated(summary = this)) {
+                            first.toNewlyCreatedTransferableAsset(first.newlyCreatedMetadata(summary = this), aggregateAmount)
+                        } else {
+                            first.toTransferableAsset(involvedAssets, aggregateAmount)
+                        }
+                    }
+
+                    is ResourceIndicator.NonFungible -> {
+                        val nonFungibleLocalIds = resources.filterIsInstance<ResourceIndicator.NonFungible>()
+                            .fold(setOf<NonFungibleLocalId>(), operation = { acc, resource ->
+                                acc + resource.indicator.nonFungibleLocalIds.toSet()
+                            })
+                        val resource = first.copy(indicator = NonFungibleResourceIndicator.ByIds(nonFungibleLocalIds.toList()))
+                        if (resource.isNewlyCreated(summary = this)) {
+                            resource.toNewlyCreatedTransferableAsset(first.newlyCreatedMetadata(summary = this))
+                        } else {
+                            resource.toTransferableAsset(involvedAssets)
+                        }
+                    }
+                }
+            }
```

Comment:
this code aggregates multiple entries for same resources into one resource item on the UI. For fungibles that mean just summing total amount, for non fungibles combining all local ids under one resource. Aggregation is set to false in code, so we do not aggregate withdrawals for now

Reply:
more or less I understand what you explained, but I do not understand why then you have to add this boolean/check there if it is always false. I mean is there any reason we need to ship this release with this bool there? Can't we do it later?

Also, in which scenario the `aggregateByResourceAddress` is true?

Reply:
I tested iOS app and it seem they do not aggregate those values, this is why there is no scenario in which this flag is true for now. But this aggregate logic was present in android app, although implemented in different place. I don't think we had a conclusion with iOS what display is correct, this is why I left this code here